controllers/mustgather/mustgather_controller.go

```diff
@@ -399,6 +385,29 @@ func (r *MustGatherReconciler) getJobFromInstance(instance *mustgatherv1alpha1.M
 	return unstructuredJob, nil
 }
 
+func (r *MustGatherReconciler) getClusterVersionForJobTemplate(name string) (string, error) {
+	clusterVersion := &configv1.ClusterVersion{}
+	err := r.GetClient().Get(context.TODO(), types.NamespacedName{Name: name}, clusterVersion)
+	if err != nil {
+		return "", fmt.Errorf("unable to get clusterversion '%v': %w", name, err)
+	}
+
+	var version string
+	for _, history := range clusterVersion.Status.History {
+		if history.State == "Completed" {
+			version = history.Version
+			break
+		}
+	}
+	if version == "" {
+		return "", goerror.New("unable to determine cluster version from status history")
+	}
+
+	version = version[:strings.LastIndex(version, ".")]
```

Comment:
Can we unit test this function? Small edge case, but I think this doesn't work when the version is something like `4.14.0-rc.0`

You can use https://github.com/openshift/managed-cluster-validating-webhooks/blob/f0862cb7bf1869b00e1ab857ee08fcf6fc7f156c/pkg/webhooks/podimagespec/podimagespec_test.go as an example if you haven't setup a fake kube client before.